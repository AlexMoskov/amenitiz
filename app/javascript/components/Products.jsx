import React from "react";
import { Link } from "react-router-dom";
import apiService from '../services/apiService';
import CurrencyFormat from 'react-currency-format';

class Products extends React.Component {
    constructor(props) {
        super(props);
        this.state = {
            products: [],
            message: null,
            error: null
        };
    }

    componentDidMount() {
        apiService.getProducts()
            .then(response => this.setState({ products: response }))
            .catch(() => this.props.history.push("/"));
    }

    addToCart(product_id) {
        apiService.addToCart(product_id)
            .then(response => {
                this.setState({ message: true, error: null })
            })
            .catch(error => console.log(error.message));
    }

    render() {
        const { products, message } = this.state;
        const allProducts = products.map((product, index) => (
            <div key={index} className="col-md-6 col-lg-4">
                <div className="card mb-4">
                    <div className="card-body">
                        <h5 className="card-title">
                            <span className='left'>{product.name}</span>
                            <span className='right'><CurrencyFormat value={product.price} displayType={'text'} thousandSeparator={true} suffix={' â‚¬'} /></span>
                        </h5>
                        <br/>
                        <Link to={`/products/${product.id}`} className="btn custom-button left">
                            View Product
                        </Link>
                        <a className="btn btn-success right" onClick={() => this.addToCart(product.id)}>Add to cart</a>
                    </div>
                </div>
            </div>
        ));

        return (
            <React.Fragment>
                <section className="jumbotron jumbotron-fluid text-center">
                    <div className="container py-5">
                        <h1 className="display-4">Products for every occasion</h1>
                        <p className="lead text-muted">
                            We have pulled together our most popular products for you.
                        </p>
                    </div>
                </section>
                <div className="py-5">
                    <main className="container">
                        {message && <div className="alert alert-success alert-dismissible fade show" role="alert">
                            The product has been successfully added to your cart.
                        </div>}

                        <div className="row">
                            { allProducts }
                        </div>
                        <Link to="/" className="btn btn-lg btn-secondary">
                            Back home
                        </Link>
                        <div className="text-right mb-3 right">
                            <Link to="/cart" className="btn btn-lg btn-info">
                                Got to cart
                            </Link>
                        </div>
                    </main>
                </div>
            </React.Fragment>
        );
    }
}
export default Products;