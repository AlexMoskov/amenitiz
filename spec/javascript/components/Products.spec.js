import React from 'react'
import axios from 'axios'
import Adapter from 'enzyme-adapter-react-16';
import { shallow, configure } from 'enzyme'
import toJson from 'enzyme-to-json'
import Products from 'components/Products'

configure({ adapter: new Adapter() });
jest.mock('axios')

describe('Products', () => {
    const props = {}
    const component = shallow(<Products {...props} />)

    it('renders Products component', () => {
        expect(toJson(component)).toMatchSnapshot()
    })
})
