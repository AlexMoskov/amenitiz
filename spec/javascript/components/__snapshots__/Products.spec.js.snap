// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Products renders Products component 1`] = `
<Fragment>
  <section
    className="jumbotron jumbotron-fluid text-center"
  >
    <div
      className="container py-5"
    >
      <h1
        className="display-4"
      >
        Products for every occasion
      </h1>
      <p
        className="lead text-muted"
      >
        We have pulled together our most popular products for you.
      </p>
    </div>
  </section>
  <div
    className="py-5"
  >
    <main
      className="container"
    >
      <div
        className="row"
      />
      <Link
        className="btn btn-lg btn-secondary"
        to="/"
      >
        Back home
      </Link>
      <div
        className="text-right mb-3 right"
      >
        <Link
          className="btn btn-lg btn-info"
          to="/cart"
        >
          Got to cart
        </Link>
      </div>
    </main>
  </div>
</Fragment>
`;
